rust/azure_iot_operations_services/src/azure_device_registry.rs

```diff
@@ -0,0 +1,1049 @@
+    AssetDatasetSchemaElementSchema, AssetEndpointProfile as GenAssetEndpointProfile,
+    AssetEndpointProfileSpecificationSchema as GenAssetEndpointProfileSpecificationSchema,
+    AssetEndpointProfileStatus as GenAssetEndpointProfileStatus,
+    AssetEndpointProfileUpdateEventTelemetry as GenAssetEndpointProfileUpdateEventTelemetry,
+    AssetEventObservabilityModeSchema, AssetEventSchemaElementSchema,
+    AssetSpecificationSchema as GenAssetSpecificationSchema, AssetStatus as GenAssetStatus,
+    AssetUpdateEventTelemetry as GenAssetUpdateEventTelemetry,
+    AuthenticationSchema as GenAuthenticationSchema, DatasetsSchemaSchemaElementSchema,
+    DetectedAsset as GenDetectedAsset, DetectedAssetDataPointSchemaElementSchema,
+    DetectedAssetDatasetSchemaElementSchema, DetectedAssetEventSchemaElementSchema,
+    DetectedAssetResponseStatusSchema, Error as GenError, EventsSchemaSchemaElementSchema,
+    MessageSchemaReference as GenMessageSchemaReference, MethodSchema, RetainSchema,
+    Topic as GenTopic, UsernamePasswordCredentialsSchema as GenUsernamePasswordCredentialsSchema,
+    X509credentialsSchema as GenX509credentialsSchema,
+};
+use crate::azure_device_registry::adr_type_gen::aep_type_service::client as adr_type_gen;
+    DiscoveredAssetEndpointProfile as GenDiscoveredAssetEndpointProfile,
+    DiscoveredAssetEndpointProfileResponseStatusSchema,
+    SupportedAuthenticationMethodsSchemaElementSchema,
+};
+
+/// Azure Device Registry generated code
+mod adr_name_gen;
+mod adr_type_gen;
+/// Azure Device Registry Client implementation wrapper
+mod client;
+
+pub use client::Client;
+
+/// Represents an error that occurred in the Azure Device Registry Client implementation.
+#[derive(Debug, Error)]
+#[error(transparent)]
+pub struct Error(#[from] ErrorKind);
+
+impl Error {
+    /// Returns the [`ErrorKind`] of the error.
+    #[must_use]
+    pub fn kind(&self) -> &ErrorKind {
+        &self.0
+    }
+}
+
+/// Represents the kinds of errors that occur in the Azure Device Registry Client implementation.
+#[derive(Error, Debug)]
+#[allow(clippy::large_enum_variant)]
+pub enum ErrorKind {
+    /// An error occurred in the AIO Protocol. See [`AIOProtocolError`] for more information.
+    #[error(transparent)]
+    AIOProtocolError(#[from] AIOProtocolError),
+    /// An error occurred during serialization of a request.
+    #[error("{0}")]
+    SerializationError(String),
+    /// An argument provided for a request was invalid.
+    #[error("{0}")]
+    InvalidArgument(String),
+    /// An error was returned by the Azure Device Registry Service.
+    #[error("{0:?}")]
+    ServiceError(ServiceError),
+    /// A aep or an asset may only have one observation at a time.
+    #[error("Aep or asset may only be observed once at a time")]
+    DuplicateObserve,
+    /// A aep or an asset had an error during observation.
+    #[error("{0}")]
+    ObservationError(String),
```

Comment:
Why is this a string? Unless I missed something, the string is always identical.